deploying-a-django-app-on-okteto.md

```diff
@@ -0,0 +1,704 @@
+# Deploying a Django App on Okteto
+
+## Introduction
+
+In this tutorial, We will building and deploying a Django Role based Authorization Service API(RBAS) on Okteto.
+
+An RBAS application, is a system with checks to give access to users based on their roles in the system. 
+
+What is Okteto ?
+
+Okteto is a cloud platform that enables developers develop and deploy cloud like Docker and Kubernetes applications. 
+
+What is Django?
+
+Django is one of the web framework of python for building APIs. Django is a framework that comes with out of the box functionality used in developing web apps and API, making your development faster and easier.
+
+
+## Prerequisites 
+- Knowledge of building APIs
+- Basic Knowledge of Django
+- Basic Knowledge of Cloud Infrastructure 
+
+## Scaffold Project
+
+To start building, we will first create and activate a virtual environment for our project:
+
+
+    $ python3 -m venv venv
+    $ source venv/bin/activate
+
+Next, we install Django:
+
+
+    $ pip install django
+
+Running `django-admin`. 
+
+
+    $ django-admin startproject simple_rbas
+
+`django-admin` is a command provided by Django to help bootstrap your project, creating essential directories and files to get started.
+
+Navigate into the new project created;
+
+
+    $ cd simple_rbas
+
+Create `requirements.txt` file in the root folder and add the following packages:
+
+
+    django
+    djangorestframework
+    django-environ
+    psycopg2-binary
+    gunicorn
+
+Install the packages using the `pip` command:
+
+
+    pip install -r requirements.txt 
+## Development
+
+In this section, we will be building up our model, serializers, permissions for our views, and routing to our views.
+
+We will also be setting Docker and Okteto for our project.
+
+### Setting Up Project
+
+To get started with building our app, we’ll create an app inside our project.  Run this command inside your project directory;
+
+
+    python manage.py startapp users
+
+ 
+ This command creates a new directory named `users` in your project directory, looking like this:
+ 
+
+    users
+      - migrations
+        -- __init__.py
+      -- __init__.py
+      -- admin.py
+      -- apps.py
+      -- models.py
+      -- tests.py
+      -- views.py
+
+For our authentication, we will be using using `http-only cookies` and not `localStorage`, to set our authentication.  This is to advised to reduce the risk of security issues.
+
+We will be creating an authentication middleware and  for the project;
```

Comment:
```suggestion
We will be creating an authentication middleware  for the project;
```

Reply:
Could you add a small paragraph before explaining why we are using middleware, how that's a common pattern for Django apps, or something like that?